macaroons/store.go

```diff
@@ -215,3 +243,13 @@ func (r *RootKeyStorage) Close() error {
 	}
 	return r.Backend.Close()
 }
+
+// getRootKeyID retrieves the root key ID from context using the key
+// RootKeyIDContextKey.
+func getRootKeyID(ctx context.Context) ([]byte, error) {
```

Comment:
We should export this function and perhaps call it `RootKeyIDFromContext`. And to make it feature complete, also add a `ContextWithRootKeyID`? Could be used in `NewMacaroon` and the unit tests as well.